src/Nancy/Json/Simple/NancySerializationStrategy.cs

```diff
@@ -59,20 +83,34 @@ public void RegisterConverters(IEnumerable<JavaScriptPrimitiveConverter> javaScr
 
         protected override string MapClrMemberNameToJsonFieldName(string clrPropertyName)
         {
-            return this.retainCasing ? base.MapClrMemberNameToJsonFieldName(clrPropertyName) :
-                clrPropertyName.ToCamelCase();
+            return this.retainCasing
+                ? base.MapClrMemberNameToJsonFieldName(clrPropertyName)
+                : clrPropertyName.ToCamelCase();
         }
 
+        /// <summary>
+        /// Deserialize an object
+        /// </summary>
+        /// <param name="value">The object to deserialize</param>
+        /// <param name="type">The type of object to deserialize</param>
+        /// <returns></returns>
         public override object DeserializeObject(object value, Type type)
         {
-            if (type.IsEnum || (ReflectionUtils.IsNullableType(type) && Nullable.GetUnderlyingType(type).IsEnum))									
-                return value == null ? null : Enum.Parse(ReflectionUtils.IsNullableType(type) ? Nullable.GetUnderlyingType(type) : type, value.ToString(), true);
+            if (type.IsEnum || (ReflectionUtils.IsNullableType(type) && Nullable.GetUnderlyingType(type).IsEnum))
+            {
+                var typeToParse = ReflectionUtils.IsNullableType(type) ? Nullable.GetUnderlyingType(type) : type;
+
+                return value == null
+                    ? null
+                    : Enum.Parse(typeToParse, value.ToString(), true);
+            }
 
             var primitiveConverter = this.FindPrimitiveConverter(type);
             if (primitiveConverter != null)
             {
                 return primitiveConverter.Deserialize(value, type);
             }
+
             var valueDictionary = value as IDictionary<string, object>;
             if (valueDictionary != null)

```

Comment:
From this point forward, the nesting gets really deep. Maybe we should invert the `if` conditions and flatten out the nesting. We'd make the call to `base.DeserializeObject(value, type)` from two more places, but we'd flatten it right out and only have 1-level conditions in the entire method